### PR TITLE
Provided vm creation details to user in non-verbose mode

### PR DESCRIPTION
**Pull request by dheerajd-msys**

### CODE REVIEW

File: lib/azure/resource_management/ARM_interface.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ module Azure
   class ResourceManagement
     class ARMInterface < AzureInterface
       include Azure::ARM::ARMBase
-
+
```
Comment (Aliasgar16) [inline lib/azure/resource_management/ARM_interface.rb]: Remove white spaces.

File: lib/azure/resource_management/ARM_interface.rb [hunk 2]
```diff
@@ -250,10 +250,11 @@ def create_server(params = {})
         platform(params[:azure_image_reference_offer])
         ## resource group creation
         if resource_group_exist?(params[:azure_resource_group_name])
-          Chef::Log.info("Resource Group #{params[:azure_resource_group_name]} already exist. Skipping its creation.")
-          Chef::Log.info("Adding new VM #{params[:azure_vm_name]} to this resource group.")
+          ui.log("Resource Group #{params[:azure_resource_group_name]} already exist. Skipping its creation.")
+          ui.log("Adding new VM #{params[:azure_vm_name]} to this resource group.")
         else
-          Chef::Log.info("Creating ResourceGroup....")
+          ui.log("Creating ResourceGroup.... #{("\n")} ")
```
Comment (Aliasgar16) [inline lib/azure/resource_management/ARM_interface.rb]: Can we remove the additional #{("\n")} string from the log message string and directly write the character "\n" in the log message string if it serves the purpose.

File: lib/azure/resource_management/ARM_interface.rb [hunk 3]
```diff
@@ -376,7 +377,9 @@ def create_virtual_machine(params)
         begin
           virtual_machine = compute_management_client.virtual_machines.create_or_update(params[:azure_resource_group_name], vm_params.name, vm_params).value!.body
         rescue Exception => e
-          Chef::Log.error("Failed to create the Virtual Machine -- exception being rescued: #{e.to_s}")
+        ui.log("Failed to create the Virtual Machine, use verbose mode for more details")          
+        Chef::Log.error("Failed to create the Virtual Machine -- exception being rescued: #{e.to_s}")
```
Comment (Aliasgar16) [inline lib/azure/resource_management/ARM_interface.rb]: Incorrect indentation.

File: lib/azure/resource_management/ARM_interface.rb [hunk 4]
```diff
@@ -385,20 +388,22 @@ def create_virtual_machine(params)
...
           params[:azure_resource_group_name]
         )
 
-        Chef::Log.info("StorageAccount creation successfull.")
         virtual_hard_disk = get_vhd(
           params[:azure_storage_account],
           params[:azure_os_disk_name]
         )
 
+        Chef::Log.info("StorageAccount creation successfull.")
```
Comment (Aliasgar16) [inline lib/azure/resource_management/ARM_interface.rb]: `Chef::Log.info` not replaced here with `ui.log` and other thing is why this message is relocated here from its original place.

File: lib/azure/resource_management/ARM_interface.rb [hunk 5]
```diff
@@ -474,25 +479,24 @@ def subnet_exist?(resource_group_name, vnet_name, subnet_name)
       def create_network_profile(params)
         if vnet_exist?(params[:azure_resource_group_name], params[:azure_vnet_name])
           vnet = network_resource_client.virtual_networks.get(params[:azure_resource_group_name], params[:azure_vnet_name]).value!.body
-          Chef::Log.info("Found existing vnet #{vnet.name}...")
+          Chef::Log.info("Found existing vnet #{vnet.name}...")
```
Comment (Aliasgar16) [inline lib/azure/resource_management/ARM_interface.rb]: Remove white spaces.

File: lib/azure/resource_management/ARM_interface.rb [hunk 6]
```diff
@@ -474,25 +479,24 @@ def subnet_exist?(resource_group_name, vnet_name, subnet_name)
...
             params[:azure_vnet_name],
             params[:azure_service_location]
           )
           Chef::Log.info("VirtualNetwork creation successfull.")
         end
-
-        Chef::Log.info("Virtual Network name is: #{vnet.name}")
-        Chef::Log.info("Virtual Network ID is: #{vnet.id}")
+          Chef::Log.info("Virtual Network name is: #{vnet.name}")        
+          Chef::Log.info("Virtual Network ID is: #{vnet.id}")
```
Comment (Aliasgar16) [inline lib/azure/resource_management/ARM_interface.rb]: Incorrect indentation.

File: lib/azure/resource_management/ARM_interface.rb [hunk 7]
```diff
@@ -474,46 +467,50 @@ def subnet_exist?(resource_group_name, vnet_name, subnet_name)
...
+            Chef::Log.info("Virtual Network ID is: #{vnet.id}")
 
         if subnet_exist?(params[:azure_resource_group_name], vnet.name, params[:azure_vnet_subnet_name])
           sbn = network_resource_client.subnets.get(params[:azure_resource_group_name], vnet.name, params[:azure_vnet_subnet_name]).value!.body
-          Chef::Log.info("Found subnet #{sbn.name} under virtual network #{vnet.name} ...")
-        else
-          Chef::Log.info("Creating Subnet....")
-          sbn = create_subnet(
+            
+            Chef::Log.info("Found subnet #{sbn.name} under virtual network #{vnet.name} ...")
```
Comment (NimishaS) [inline lib/azure/resource_management/ARM_interface.rb]: Please fix the indentation

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

**Pull request merged**